### PR TITLE
initial PR

### PR DESCRIPTION
**Pull request by doodyparizada**
logic implementation for create wallet and submit payment.

### CODE REVIEW

File: payment/__init__.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+
+app = Flask(__name__)
+
+
+@app.route('/wallets/', methods=['POST'])
+@handle_errors
+def create_wallet():
+    body = request.get_json()
+    wallet_address = body['wallet_address']
```
Comment (bolshoy) [inline payment/__init__.py]: check the params? Will throw exception if `wallet_address` does not exist

File: payment/__init__.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    # wallet creation is idempotent - no locking needed
+    blockchain.create_wallet(wallet_address)
```
Comment (bolshoy) [inline payment/__init__.py]: if this account already exists, it will throw exception `op_already_exists`

File: payment/__init__.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+    return jsonify(), 201
+
+
+@app.route('/orders/<order_id>', methods=['GET'])
+@handle_errors
+def get_order(order_id):
+    t = Transaction.get_by_order_id(order_id)
```
Comment (bolshoy) [inline payment/__init__.py]: if no such order, you need to translate it to 404

File: payment/__init__.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    return jsonify(t.to_primitive())
+
+
+@app.route('/orders/', methods=['POST'])
+@handle_errors
+def pay():
```
Comment (bolshoy) [inline payment/__init__.py]: POST to `/orders` usually creates a new order

File: payment/__init__.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    body = request.get_json()
+    order_id = body['order_id']
+    app_id = body['app_id']
+    wallet_address = body['wallet_address']
+    amount = body['amount']
```
Comment (bolshoy) [inline payment/__init__.py]: check each one and throw 400 on error

File: payment/__init__.py [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+    with lock('payment:{}'.format(order_id)):
+        try:
+            t = Transaction.get_by_order_id(order_id)
+            return jsonify(t.to_primitive())
+        except KeyError:
+            pass
+
+        t = blockchain.pay_to(wallet_address, amount, app_id, order_id)
+        t.save()
```
Comment (bolshoy) [inline payment/__init__.py]: any transaction context required?

File: payment/blockchain.py [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    base_seed=config.STELLAR_BASE_SEED,
+    horizon_endpoint_uri=config.STELLAR_HORIZON_URL,
+    network=config.STELLAR_NETWORK,
+    channel_seeds=config.STELLAR_CHANNEL_SEEDS)
+
+
+def create_wallet(public_address: str) -> None:
+    """create a wallet."""
+    initial_xlm_amount = config.STELLAR_INITIAL_XLM_AMOUNT
+    tx_id = kin_sdk.create_account(public_address, starting_balance=initial_xlm_amount)
```
Comment (bolshoy) [inline payment/blockchain.py]: here you can catch `op_already_exists ` and just return

File: payment/blockchain.py [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+    print("received tx_id: {}".format(tx_id))
+
+
+def pay_to(public_address: str, amount: int, app_id: str, order_id: str) -> Transaction:
+    '''send kins to an address'''
+    print('sending kin to address: {}'.format(public_address))
+    memo = Transaction.create_memo(app_id, order_id)
```
Comment (bolshoy) [inline payment/blockchain.py]: `create_memo` is too app-specific, not a part of Transaction

File: payment/config.py [hunk 9]
```diff
@@ -0,0 +1,9 @@
+import os
+
+
+STELLAR_HORIZON_URL = 'https://horizon-testnet.stellar.org/'
+STELLAR_NETWORK = 'TESTNET'
+STELLAR_KIN_ISSUER_ADDRESS = 'GCKG5WGBIJP74UDNRIRDFGENNIH5Y3KBI5IHREFAJKV4MQXLELT7EX6V'
+STELLAR_CHANNEL_SEEDS = os.environ['STELLAR_CHANNEL_SEEDS'].split(',')
```
Comment (bolshoy) [inline payment/config.py]: are you going to get it from KMS in the future?

Comment (doodyparizada) [reply payment/config.py]: yes/ or some other wrapping tool will get it and pass it on

File: payment/__init__.py (hunk 4 above)
Comment (doodyparizada) [reply payment/__init__.py]: It's in the process of creating - returing 202

File: payment/__init__.py (hunk 6 above)
Comment (doodyparizada) [reply payment/__init__.py]: what does this mean?

**Pull request merged**